src/feature/flow_chart_panel/components/WatchBtn.tsx

```diff
@@ -0,0 +1,66 @@
+import { useFlowChartState } from "@src/hooks/useFlowChartState";
+import { useEffect, useState } from "react";
+import { Node, Edge } from "reactflow";
+import { ElementsData } from "flojoy/types";
+import { useFlowChartGraph } from "@src/hooks/useFlowChartGraph";
+import { Label } from "@src/components/ui/label";
+import { Switch } from "@src/components/ui/switch";
+import {
+  Tooltip,
+  TooltipProvider,
+  TooltipTrigger,
+  TooltipContent,
+} from "@/components/ui/tooltip";
+
+interface WatchBtnProps {
+  playFC: (nodes: Node<ElementsData>[], edges: Edge[]) => void;
+  cancelFC: () => void;
+}
+
+const WatchBtn = ({ playFC, cancelFC }: WatchBtnProps) => {
+  const { nodeParamChanged } = useFlowChartState();
+  const { nodes, edges } = useFlowChartGraph();
+  const [isWatching, setIsWatching] = useState(false);
+  const [runOnceRightAfterToggle, setRunOnceRightAfterToggle] = useState(false);
```

Comment:
Is this needed? Can't you just have a `useEffect` that runs only when `isWatching` changes?